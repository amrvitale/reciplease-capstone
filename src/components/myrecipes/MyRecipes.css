* {
    box-sizing: border-box;
  }

  .container {
    display: flex;
    flex-direction: row;
    flex-flow: row wrap;
    align-items: center;
    justify-content: space-around;
  }

  .card { 
    margin: 1px;
    padding: 5px;
    border: 1px solid #2B2D42;
    border-radius: 5px;
    background-color: rgb(255, 213, 136);
    justify-content: space-evenly;
  }

  .container .card {
    padding: 3px;
    display: flex;
    border: 1px #2B2D42;  
}

li.card{
    margin: 30px;
    justify-self: center;
    height: 250px;
    width: 320px;
    box-shadow: 3px 3px 5px 6px #ccc;
}

.myRecipesHeading {
    font-family: 'Amaranth', sans-serif;;
    color: white;
    text-shadow:
    4px 4px 0 black,
    -1px -1px 0 #000,  
     1px -1px 0 #000,
     -1px 1px 0 #000,
      1px 1px 0 #000;
}

.myRecipesPageHero{
    background-image: url(./myRecipes.jpg);
    background-position: center; /* Center the image */
    background-repeat: no-repeat; /* Do not repeat the image */
    background-size: cover; /* Resize the background image to cover the entire container */
}