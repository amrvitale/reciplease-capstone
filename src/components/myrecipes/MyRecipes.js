import React from 'react';
import Recipe from '../recipe/Recipe';
import './MyRecipes.css';

/* This component renders the My Recipes page, which renders the Recipe component. Users get here by clicking My Kitchen > View My Recipes. */

class MyRecipes extends React.Component {
    render() {
        console.log(this.props.recipes)
        return (
            <div className="myRecipesPage">
                <div className="myRecipesPageHero">
                    <h1 className="myRecipesHeading">My Recipes</h1>
                </div>
                <ul className="container">
                    {this.props.recipes.map(recipe =>
                        <li className="card" key={recipe.id}>
                            <Recipe 
                                id={recipe.id}
                                recipename={recipe.recipename}
                                experience={recipe.experience}
                                preptime={recipe.preptime}
                                cooktime={recipe.cooktime}
                                servings={recipe.servings}
                                ingredients={recipe.ingredients}
                                directions={recipe.directions}
                            />
                        </li>
                    )}
                </ul>
            </div>
        );
    }
}

export default MyRecipes;