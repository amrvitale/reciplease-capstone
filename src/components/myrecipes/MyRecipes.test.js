import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MyRecipes from './MyRecipes'

describe (`MyRecipes component`, () => {
    it ('renders a .MyRecipes by default', () => {
        const wrapper = shallow( <MyRecipes recipes={[]}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});