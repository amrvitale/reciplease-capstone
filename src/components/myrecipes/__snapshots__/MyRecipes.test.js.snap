// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipes component renders a .MyRecipes by default 1`] = `
<div
  className="myRecipesPage"
>
  <div
    className="myRecipesPageHero"
  >
    <h1
      className="myRecipesHeading"
    >
      My Recipes
    </h1>
  </div>
  <ul
    className="container"
  />
</div>
`;
